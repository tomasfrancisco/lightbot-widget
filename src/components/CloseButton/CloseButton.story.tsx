import { storiesOf } from "@storybook/react";
import * as React from "react";
import { ThemeProviderDecorator } from "../../modules/ThemeProvider";
import { CloseButton } from "./CloseButton";

storiesOf("CloseButton", module)
  .addDecorator(ThemeProviderDecorator)
  .add("default", () => <CloseButton>X</CloseButton>);
