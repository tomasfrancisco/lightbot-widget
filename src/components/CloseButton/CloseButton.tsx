import * as React from "react";
import styled from "styled-components";

export const CloseButton = styled<any, "button">("button")`
  border: none;
  outline: none;
  width: 30px;
  height: 30px;
  border-radius: 100%;
  background-color: ${props => props.theme.closeButtonBackgroundColor || "rgba(0, 0, 0, 0.2)"};
  padding-bottom: 3px;
  color: ${props => props.theme.closeButtonColor || "#fff"};
  transition: all 0.3s ease-in-out;
  cursor: pointer;

  &:hover {
    box-shadow: 0px 0px 4px #000;
  }
`;
