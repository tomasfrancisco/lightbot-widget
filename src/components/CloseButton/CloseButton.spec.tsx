import { CloseButton } from "components/CloseButton/CloseButton";
import { shallow } from "enzyme";
import * as React from "react";

describe("modules/Widget/Window", () => {
  let component;

  beforeEach(() => {
    component = shallow(<CloseButton />);
  });

  it("renders", () => {
    expect(component).toMatchSnapshot();
  });
});
