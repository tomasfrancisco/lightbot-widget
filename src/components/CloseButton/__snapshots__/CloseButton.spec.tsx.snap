// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modules/Widget/Window renders 1`] = `
<button
  className="sc-bdVaJa iqOGyR"
/>
`;
